RuntimeException running vaadin:upgrade8

```
Exception in thread "main" java.lang.RuntimeException: org.eclipse.aether.resolution.DependencyResolutionException: Failed to read artifact descriptor for com.vaadin:vaadin-compatibility-server:jar:8.0.0
	at com.vaadin.framework8.migrate.MavenResolver.resolve(MavenResolver.java:73)
	at com.vaadin.framework8.migrate.VadinJarFinder.download(VadinJarFinder.java:55)
	at com.vaadin.framework8.migrate.VadinJarFinder.lambda$get$1(VadinJarFinder.java:22)
	at java.util.Optional.orElseGet(Optional.java:267)
	at com.vaadin.framework8.migrate.VadinJarFinder.get(VadinJarFinder.java:22)
	at com.vaadin.framework8.migrate.Migrate.main(Migrate.java:39)
Caused by: org.eclipse.aether.resolution.DependencyResolutionException: Failed to read artifact descriptor for com.vaadin:vaadin-compatibility-server:jar:8.0.0
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:387)
	at com.vaadin.framework8.migrate.MavenResolver.resolve(MavenResolver.java:68)
	... 5 more
Caused by: org.eclipse.aether.collection.DependencyCollectionException: Failed to read artifact descriptor for com.vaadin:vaadin-compatibility-server:jar:8.0.0
	at org.eclipse.aether.internal.impl.DefaultDependencyCollector.collectDependencies(DefaultDependencyCollector.java:208)
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:341)
	... 6 more
Caused by: org.eclipse.aether.resolution.ArtifactDescriptorException: Failed to read artifact descriptor for com.vaadin:vaadin-compatibility-server:jar:8.0.0
	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:282)
	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor(DefaultArtifactDescriptorReader.java:198)
	at org.eclipse.aether.internal.impl.DefaultDependencyCollector.collectDependencies(DefaultDependencyCollector.java:202)
	... 7 more
Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: Could not transfer artifact com.vaadin:vaadin-compatibility-server:pom:8.0.0 from/to central (http://central.maven.org/maven2/): /usr/share/maven/repository/com/vaadin/vaadin-compatibility-server/8.0.0/vaadin-compatibility-server-8.0.0.pom.part.lock (No such file or directory)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:444)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:246)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact(DefaultArtifactResolver.java:223)
	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:267)
	... 9 more
Caused by: org.eclipse.aether.transfer.ArtifactTransferException: Could not transfer artifact com.vaadin:vaadin-compatibility-server:pom:8.0.0 from/to central (http://central.maven.org/maven2/): /usr/share/maven/repository/com/vaadin/vaadin-compatibility-server/8.0.0/vaadin-compatibility-server-8.0.0.pom.part.lock (No such file or directory)
	at org.eclipse.aether.connector.basic.ArtifactTransportListener.transferFailed(ArtifactTransportListener.java:43)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run(BasicRepositoryConnector.java:355)
	at org.eclipse.aether.util.concurrency.RunnableErrorForwarder$1.run(RunnableErrorForwarder.java:67)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$DirectExecutor.execute(BasicRepositoryConnector.java:581)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector.get(BasicRepositoryConnector.java:249)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads(DefaultArtifactResolver.java:520)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:421)
	... 12 more
Caused by: java.io.FileNotFoundException: /usr/share/maven/repository/com/vaadin/vaadin-compatibility-server/8.0.0/vaadin-compatibility-server-8.0.0.pom.part.lock (No such file or directory)
	at java.io.RandomAccessFile.open0(Native Method)
	at java.io.RandomAccessFile.open(RandomAccessFile.java:316)
	at java.io.RandomAccessFile.<init>(RandomAccessFile.java:243)
	at org.eclipse.aether.connector.basic.PartialFile$LockFile.tryLock(PartialFile.java:123)
	at org.eclipse.aether.connector.basic.PartialFile$LockFile.lock(PartialFile.java:71)
	at org.eclipse.aether.connector.basic.PartialFile$LockFile.<init>(PartialFile.java:58)
	at org.eclipse.aether.connector.basic.PartialFile$Factory.newInstance(PartialFile.java:210)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$GetTaskRunner.runTask(BasicRepositoryConnector.java:432)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run(BasicRepositoryConnector.java:350)
	... 17 more
Migration to Vaadin 8 failed
org.codehaus.mojo.gwt.shell.JavaCommandException: Command [[
/bin/sh -c '/home/javydreamercsw/Documents/jdk1.8.0_121/jre/bin/java' '-Xmx512M' '-Xss1024k' '-classpath' '/home/javydreamercsw/.m2/repository/com/vaadin/framework8-migration-tool/8.0-SNAPSHOT/framework8-migration-tool-8.0-SNAPSHOT.jar' 'com.vaadin.framework8.migrate.Migrate' '-version=8.0.0'
]] failed with status 1
	at org.codehaus.mojo.gwt.shell.JavaCommand.execute(JavaCommand.java:327)
	at com.vaadin.integration.maven.Vaadin8UpgradeMojo.doExecute(Vaadin8UpgradeMojo.java:85)
	at org.codehaus.mojo.gwt.shell.AbstractGwtShellMojo.execute(AbstractGwtShellMojo.java:182)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
```

Is the problem that it is trying to write the downloaded files to the wrong folder, i.e. uses the wrong Maven home? 
```
Caused by: java.io.FileNotFoundException: /usr/share/maven/repository/com/vaadin/vaadin-compatibility-server/8.0.0/vaadin-compatibility-server-8.0.0.pom.part.lock (No such file or directory)
```

Should this be 
```
/home/javydreamercsw/.m2/repository/com/vaadin/vaadin-compatibility-server/8.0.0/vaadin-compatibility-server-8.0.0.pom.part.lock
```
 or do you have some kind of special setup?

--- a/src/main/java/com/vaadin/framework8/migrate/VadinJarFinder.java
+++ b/src/main/java/com/vaadin/framework8/migrate/VadinJarFinder.java
@@ -11,10 +11,7 @@ public class VadinJarFinder {
 
     public static String get(String moduleName, String version)
             throws IOException {
-        String m2 = System.getenv("M2_HOME");
-        if (m2 == null) {
-            m2 = System.getenv("HOME") + "/.m2";
-        }
+        String m2 = System.getenv("HOME") + "/.m2";
 
         final String m2repositoryFolder = m2 + "/repository";
 

